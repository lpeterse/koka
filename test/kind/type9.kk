// test name clash on type synonyms
alias id = forall<a> a -> a
alias tt  = forall<a> a -> id
alias choose = id -> id
