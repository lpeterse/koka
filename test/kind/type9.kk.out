kind/type9/choose: V
kind/type9/id: V
kind/type9/tt: V
public alias choose = (id) -> id = 2
// test name clash on type synonyms
public alias id = forall<a> (a) -> a = 1
public alias tt = forall<a> (a) -> id = 2